setup.py

```diff
@@ -118,10 +118,19 @@ def build_or_install_bokehjs(packages: list[str]) -> None:
         raise ValueError(f"Unrecognized action {action!r}")
     print(f"Used {bright(yellow(kind))} BokehJS from {loc}\n")
 
+def check_tags() -> None:
+    try:
+        tags = subprocess.check_output(("git", "tag", "-l")).split()
+        if len(tags) < 5: # arbitrary "too-low" cutoff, there will always be more than 5 tags, if there are any
+            die(bright(red(MISSING_TAGS)))
+    except Exception:
+        print(bright(yellow("!!! Could not check repo tags. Please ensure full tag history")))
+
 def die(x: str) -> NoReturn:
     print(f"{x}\n")
     sys.exit(1)
 
+MISSING_TAGS = "\n!!! This repository is missing git tags! Full tag history is required to build Bokeh\n!!!\n!!! See https://docs.bokeh.org/en/latest/docs/dev_guide/setup.html#troubleshooting"  # noqa
```

Comment:
I would suggest this for readablity:
```suggestion
MISSING_TAGS = """
!!! This repository is missing git tags! Full tag history is required to build Bokeh.
!!!
!!! See https://docs.bokeh.org/en/latest/docs/dev_guide/setup.html#troubleshooting
"""
```